### PR TITLE
Stats

### PR DESCRIPTION
**Pull request by AlanKK**
Let me know what you think about this

### CODE REVIEW

File: config.sample.py [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+    # ------------------------------------------------------------------------
+
+    # how long to sleep in seconds between stats reports
+    'sleep_time': 300,
+
+    # print header every nth report
+    'header_every_nth': 20,
+
+    # write a separate .csv file for use in excel.
+    'write_csv': True,
```
Comment (Murodese) [inline config.sample.py]: Where's this writing to?

Comment (AlanKK) [inline config.sample.py]: stdout.  if you enable write_csv, it also writes to <logging_dir>/stats.csv.

logging_dir = os.path.dirname(config.log.get('logging_file'))
csv_path = os.path.join(logging_dir, "stats.csv")

Comment (AlanKK) [inline config.sample.py]: I realized I should put stats.py in scripts and update the README.  I'll do that now.

Comment (AlanKK) [inline config.sample.py]: OK, it's done with the last check in.

**Comment by Murodese** [comment]
I chucked this in the issue too, but did you want to leave this until you do the byobu layout? You'll need to edit it for the new logfile names etc anyway.

**Comment by AlanKK** [comment]
I'd like to move it into the dev branch so I can get it out of the way.  I just did a merge with your latest to pick up the logging change.  I'm getting some db errors I don't understand:

python3 scripts/stats.py 
 Parts               | Binaries            | Releases
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 951, in _execute_context
    context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/default.py", line 436, in do_execute
    cursor.execute(statement, parameters)
psycopg2.ProgrammingError: column releases.pre_id does not exist
LINE 2: ...k_id, releases.episode_id AS releases_episode_id, releases.p...
                                                             ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "scripts/stats.py", line 70, in <module>
    parts, binaries, releases = get_stats()    
  File "scripts/stats.py", line 32, in get_stats
    releases = db.query(Release).count()
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/orm/query.py", line 2640, in count
    return self.from_self(col).scalar()
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/orm/query.py", line 2426, in scalar
    ret = self.one()
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/orm/query.py", line 2395, in one
    ret = list(self)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/orm/query.py", line 2438, in __iter__
    return self._execute_and_instances(context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/orm/query.py", line 2453, in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 729, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/sql/elements.py", line 322, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 826, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 958, in _execute_context
    context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 1159, in _handle_dbapi_exception
    exc_info
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/util/compat.py", line 188, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=exc_value)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/util/compat.py", line 181, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 951, in _execute_context
    context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/default.py", line 436, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (ProgrammingError) column releases.pre_id does not exist
LINE 2: ...k_id, releases.episode_id AS releases_episode_id, releases.p...
                                                             ^
 'SELECT count(*) AS count_1 \nFROM (SELECT releases.id AS releases_id, releases.added AS releases_added, releases.posted AS releases_posted, releases.name AS releases_name, releases.search_name AS releases_search_name, releases.posted_by AS releases_posted_by, releases.status AS releases_status, releases.grabs AS releases_grabs, releases.size AS releases_size, releases.passworded AS releases_passworded, releases.unwanted AS releases_unwanted, releases.group_id AS releases_group_id, releases.category_id AS releases_category_id, releases.regex_id AS releases_regex_id, releases.tvshow_id AS releases_tvshow_id, releases.tvshow_metablack_id AS releases_tvshow_metablack_id, releases.movie_id AS releases_movie_id, releases.movie_metablack_id AS releases_movie_metablack_id, releases.nzb_id AS releases_nzb_id, releases.rar_metablack_id AS releases_rar_metablack_id, releases.nfo_id AS releases_nfo_id, releases.nfo_metablack_id AS releases_nfo_metablack_id, releases.sfv_id AS releases_sfv_id, releases.sfv_metablack_id AS releases_sfv_metablack_id, releases.episode_id AS releases_episode_id, releases.pre_id AS releases_pre_id \nFROM releases) AS anon_1' {}

**Comment by brookesy2** [comment]
I believe you need to apply the latest patch from alembic as it fixes the pre/release tables.

**Comment by Murodese** [comment]
If you're using `git pull` to update, you'll need to run `alembic upgrade head` as well. `./pynab.py update` does everything for you, though.

**Comment by AlanKK** [comment]
Thanks.  FYI - I have asked the author of byobu a  couple of questions and am still waiting to get answers.  I probably won't have much more progress on it for a few days.

**Comment by AlanKK** [comment]
More errors after merging from your dev branch and running alembic upgrade head.  What am I doing wrong?

/var/www/pynab$ alembic upgrade head
INFO  [alembic.migration] Context impl PostgresqlImpl.
INFO  [alembic.migration] Will assume transactional DDL.
INFO  [alembic.migration] Running upgrade 1cbc2db68e8 -> c5c72142b8, add missing cascades to binaries
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 951, in _execute_context
    context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/default.py", line 436, in do_execute
    cursor.execute(statement, parameters)
psycopg2.ProgrammingError: constraint "releases_sfv_id_fkey" for relation "releases" already exists


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/alembic", line 9, in <module>
    load_entry_point('alembic==0.7.3', 'console_scripts', 'alembic')()
  File "/usr/local/lib/python3.4/dist-packages/alembic/config.py", line 399, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/usr/local/lib/python3.4/dist-packages/alembic/config.py", line 393, in main
    self.run_cmd(cfg, options)
  File "/usr/local/lib/python3.4/dist-packages/alembic/config.py", line 376, in run_cmd
    **dict((k, getattr(options, k)) for k in kwarg)
  File "/usr/local/lib/python3.4/dist-packages/alembic/command.py", line 165, in upgrade
    script.run_env()
  File "/usr/local/lib/python3.4/dist-packages/alembic/script.py", line 382, in run_env
    util.load_python_file(self.dir, 'env.py')
  File "/usr/local/lib/python3.4/dist-packages/alembic/util.py", line 241, in load_python_file
    module = load_module_py(module_id, path)
  File "/usr/local/lib/python3.4/dist-packages/alembic/compat.py", line 68, in load_module_py
    module_id, path).load_module(module_id)
  File "<frozen importlib._bootstrap>", line 539, in _check_name_wrapper
  File "<frozen importlib._bootstrap>", line 1591, in load_module
  File "<frozen importlib._bootstrap>", line 596, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 1220, in load
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1448, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "alembic/env.py", line 71, in <module>
    run_migrations_online()
  File "alembic/env.py", line 64, in run_migrations_online
    context.run_migrations()
  File "<string>", line 7, in run_migrations
  File "/usr/local/lib/python3.4/dist-packages/alembic/environment.py", line 742, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/usr/local/lib/python3.4/dist-packages/alembic/migration.py", line 305, in run_migrations
    step.migration_fn(**kw)
  File "/var/www/pynab/alembic/versions/c5c72142b8_add_missing_cascades_to_binaries.py", line 26, in upgrade
    op.create_foreign_key('releases_sfv_id_fkey', 'releases', 'sfvs', ['sfv_id'], ['id'], ondelete='CASCADE')
  File "<string>", line 7, in create_foreign_key
  File "/usr/local/lib/python3.4/dist-packages/alembic/operations.py", line 748, in create_foreign_key
    **dialect_kw)
  File "/usr/local/lib/python3.4/dist-packages/alembic/ddl/impl.py", line 167, in add_constraint
    self._exec(schema.AddConstraint(const))
  File "/usr/local/lib/python3.4/dist-packages/alembic/ddl/impl.py", line 105, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 729, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/sql/ddl.py", line 69, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 783, in _execute_ddl
    compiled
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 958, in _execute_context
    context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 1159, in _handle_dbapi_exception
    exc_info
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/util/compat.py", line 188, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=exc_value)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/util/compat.py", line 181, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 951, in _execute_context
    context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/default.py", line 436, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (ProgrammingError) constraint "releases_sfv_id_fkey" for relation "releases" already exists
 'ALTER TABLE releases ADD CONSTRAINT releases_sfv_id_fkey FOREIGN KEY(sfv_id) REFERENCES sfvs (id) ON DELETE CASCADE' {}

**Comment by Murodese** [comment]
Re-pull and try again.

**Comment by AlanKK** [comment]
Much better.  thx

**Comment by Murodese** [comment]
Whoops, this is headed towards master. Can you kill the PR and remake a new one to merge with development-postgres?

**Pull request closed without merging**